File: sdk/src/main/java/org/maibot/sdk/ioc/ObjectFactory.java
package org.maibot.sdk.ioc;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 标记一个类为对象工厂，表示该类负责创建某些对象的实例。
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ObjectFactory {
}

File: sdk/src/main/java/org/maibot/sdk/ioc/ObjectFactory.kt
package org.maibot.sdk.ioc

/**
 * 标记一个类为对象工厂，表示该类负责创建某些对象的实例。
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Component(singleton = false)
annotation class ObjectFactory
